lib/upload.js

```diff
@@ -262,6 +243,33 @@ class Upload {
     }
   }
 
+  _isOnline() {
+    let isOnline = true;
+    if (typeof window !== "undefined" &&
+        "navigator" in window &&
+        window.navigator.onLine === false) {
+      isOnline = false;
+    }
+
+    return isOnline;
+  }
+
+  _shouldRetry(err) {
```

Comment:
:+1: for refactoring. Could you add JSDoc-style comments to the new functions, in a similar fashion how (some) other methods already have?